specs/Menu.md

```diff
@@ -1,731 +1,545 @@
+
+Sample usages will be give in the following section of this document [Sample code](#sample-code)
+
+### Menu
+
+The root level component serves as a simplified interface (sugar) for popup positioning and triggering.
+
+### MenuList
+
+This component is used internally by `Menu` and manages the context and layout its items.
+
+`MenuList` can also be used separately as the standalone variant of the `Menu`, since it should not control popup positioning or triggers. It is the only component in the API that can be used standalone. Envisioned to be used with more complex popup or trigger scenarios where the `Menu` component does not provide enough control for these situations.
 
-Since the Menu is only rendering its children, the only slot necessary is the one for the Root `div`. The rest can be passed in as children variants.
+### MenuSection
 
-#### MenuItem
+Creates a section inside a `MenuList`, setting up header layout and dividers between `MenuItems`.
 
-Since the MenuItem is only rendering its children, the only slot necessary is the one for the Root `div`. The rest can be passed in as children.
+The MenuSection is also a useful component to declare different selection groups (checkbox/radio) in a `MenuList`.
 
-#### SubmenuList
+### MenuItem
+
+As the name infers
 
-Since the SubmenuList is effectively the same as the Menu, it should have only one slot for the Root `div` and for all intents and purposes should behave exactly the same as the Menu slot does.
+### SubMenu
 
-#### SubmenuItem
+Creates a `Menu` component with `MenuItem` trigger and handles the positioning of the nested menu.
 
-Since the SubmenuItem is effectively the same as the MenuItem, it should have only one slot for the Root `div` and for all intents and purposes should behave exactly the same as the MenuItem slot does.
+## Sample code
 
-### Behaviors
+The below samples do not represent the definitive props of the final implemented component, but represent the ideal final implementations. Can be subject to change during the implementation phase.
 
-#### Menu
+### Default Menu
 
-The menu itself should be a list that renders menu items, it will provide context about its overall state but should not pass that directly into its items as props.
+```typescript
+const trigger = <button> Open menu </button>
 
-##### Orientation
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem title="Option 1" />
+    <MenuItem title="Option 2" />
+    <MenuItem title="Option 3" />
+  <Menu>
+)
+```
 
-The menu should provide some help for orienting its contents either vertically, like a left nav, or horizontally, like a nav bar. Ideally this will just be a prop that gets put into context so each item can decide how it should appear. It's possible that the menu should enforce direction itself.
+### Sections and submenus
+
+```typescript
+const trigger = <button> Open menu </button>
+
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem title="Option 1" />
```

Comment:
`title` collides with HTML `title` attribute. `content` or something similar might be better

Reply:
Makes sense, work is still needed to define the props a lot better. The sample code usages for now should show the intention of the API design

Reply:
After more thought, I've converted to using children here.